spotify/http.py

```diff
@@ -1699,13 +1702,28 @@ def search(  # pylint: disable=invalid-name
         return self.request(route, params=payload)
 
 
+REFRESH_TOKEN_URL = "https://accounts.spotify.com/api/token?grant_type=refresh_token&refresh_token={refresh_token}"
+
+
 class HTTPUserClient(HTTPClient):
     """HTTPClient for access to user endpoints."""
 
-    def __init__(self, token, loop=None):
-        super().__init__(client_id=None, client_secret=None, loop=loop)
-        self.bearer_info = {"access_token": token}
-        self.token = token
+    def __init__(self, client_id: str, client_secret: str, token: str = None, refresh_token: str = None, loop=None):
+        assert token or refresh_token
+        super().__init__(client_id, client_secret, loop=loop)
+        if token:
+            self.bearer_info = {"access_token": token}
+        self.refresh_token = refresh_token
 
     async def get_bearer_info(self, *_, **__):
-        return {"access_token": self.token}
+        if not self.refresh_token:
```

Comment:
Is this case worth having? Maybe we should just force passing refresh_token into User.from_token so the refresh_token is always defined.

Reply:
I agree. I don't think I know of a case when the oauth flow provides an access token without a refresh token however, if it's possible to get an access token without a refresh token from the oauth flow then it should be an optional keyword argument.

Reply:
I just left the refresh_token optional in the User.from_token function.

Can you think of a use case where someone actually call User.from_token?  If I was gonna  save/serialize a user then I'd always use from_refresh_token.